zerver/lib/test_classes.py

```diff
@@ -152,7 +153,14 @@ def client_patch(self, url: str, info: Dict[str, Any]={}, **kwargs: Any) -> Http
         encoded = urllib.parse.urlencode(info)
         django_client = self.client  # see WRAPPER_COMMENT
         self.set_http_headers(kwargs)
-        return django_client.patch(url, encoded, **kwargs)
+        result = django_client.patch(url, encoded, **kwargs)
+        if (url.startswith("/json") or url.startswith("/api/v1")):
+            content = ujson.loads(result.content)
+            url = re.sub(r"\?.*", "", url)
+            validate_against_openapi_schema(content,
+                                            url.replace("/json/", "/").replace("/api/v1/", "/"),
+                                            "patch", str(result.status_code))
```

Comment:
Let's extract a common function `self.validate_api_response_openapi(url, method result)` so we don't have 4 near-duplicates of this logic.

Reply:
Done.